src/main/java/com/github/redouane59/twitter/helpers/RequestHelper.java

```diff
@@ -49,6 +54,41 @@
     return Optional.ofNullable(result);
   }
 
+  public <T> Optional<T> uploadMedia(String url, File file, Class<T> classType) {
+    T result = null;
+    try {
+      OkHttpOAuthConsumer
+          consumer =
+          new OkHttpOAuthConsumer(TwitterClient.TWITTER_CREDENTIALS.getApiKey(), TwitterClient.TWITTER_CREDENTIALS.getApiSecretKey());
+      consumer.setTokenWithSecret(TwitterClient.TWITTER_CREDENTIALS.getAccessToken(), TwitterClient.TWITTER_CREDENTIALS.getAccessTokenSecret());
+
+      OkHttpClient client = new OkHttpClient.Builder()
+          .addInterceptor(new SigningInterceptor(consumer))
+          .build();
+
+      HttpUrl.Builder httpBuilder = HttpUrl.parse(url).newBuilder();
+      RequestBody requestBody = new MultipartBody.Builder().setType(MultipartBody.FORM)
+                                                           .addFormDataPart("media", file.toString(),
+                                                                            RequestBody.create(MediaType.parse("application/octet-stream"),
+                                                                                               file))
+                                                           .build();
+      Request request = new Request.Builder()
+          .url(httpBuilder.build())
+          .post(requestBody)
+          .build();
+
+      Response response       = client.newCall(request).execute();
+      String   stringResponse = response.body().string();
+      if (response.code() < 200 || response.code() > 299) {
+        logApiError("POST", url, stringResponse, response.code());
+      }
+      result = TwitterClient.OBJECT_MAPPER.readValue(stringResponse, classType);
```

Comment:
Did you consider using of Retrofit? It will simplify a lot all this marshaling and unmarshaling.

Reply:
I don't know it at all, let me take a look !